src/Implement/UnitTestDownloadStream.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "DownloadStream.h"
+
+// test basic usage
+TEST(DOWNLOADSTREAM_TEST, Basic) {
+  // create temporary test file
+  std::fstream newfile("test", std::fstream::out);
+  newfile << "123456";
+  newfile.close();
+
+  DownloadStream down("test", 1);
+  unsigned long pos = 0;
+  std::string binary;
+  int result;
+
+  // check ReadChunk
+  for (pos = 0; pos < 6; pos++) {
```

Comment:
Consider using `sizeof(kTestFileContent)` instead of 6, this way, we change easily change the content of the test string without changing the test.

Reply:
done.